### PR TITLE
Rework repo buttons

### LINKED ISSUE #13671: Buggy CSS when browsers are not maximised, needs more wrap-around
- Gitea version (or commit ref): 1.14.0+dev-215-gd025d84d9, official docker image
- Git version: (irrelevant)
- Operating system: (irrelevant)
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [x] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes (Screenshot below)
  - [ ] No
- Log gist: (irrelevant)

## Description

Many websites falsely assume that people run their browsers maximised, taking the entire available monitor area for themselves. I am one of these people who instead tile their browser window, making it take up only half a motinor column. The resolution of the monitor in question is 1080p.

The fix is to make all horizontal flexboxes wrap around once the window width is reached.

## Screenshots

All screenshots made at the same resolution. You can see a bit of my desktop background on some of them, and the next monitor to the right.

Bildschirmfoto_2020-11-22_12-59-54
Bildschirmfoto_2020-11-22_13-04-17
Bildschirmfoto_2020-11-22_13-04-39
Bildschirmfoto_2020-11-22_13-11-31

### PR DESCRIPTION
**Pull request by silverwind**
- Replace "New PR" and "Go to File" button with Icon Button
- Move all "Add File" actions into a dropdown button
- Remove most custom styling of clone buttons

Buttons are now all equal height, mobile layout wraps gracefully.

Fixes: https://github.com/go-gitea/gitea/issues/13671
Replaces: https://github.com/go-gitea/gitea/pull/20375

Screen Shot 2022-08-01 at 21 37 53
Screen Shot 2022-08-01 at 21 37 36
Screen Shot 2022-08-01 at 21 29 22

### CODE REVIEW

**Comment by silverwind** [comment]
Maybe should backport this into 1.17. I think all these green "Add file" buttons there are a eyesore.

**Comment by zeripath** [comment]
So unfortunately there's still a way to get this to totally break:

Screenshot from 2022-08-01 21-39-40

File: templates/repo/commits.tmpl [hunk 1]
```diff
@@ -3,9 +3,9 @@
 	{{template "repo/header" .}}
 	<div class="ui container">
 		{{template "repo/sub_menu" .}}
-		<div class="ui secondary stackable menu mobile--margin-between-items">
-		{{template "repo/branch_dropdown" dict "root" .}}
-			<div class="fitted item">
+		<div class="flex-button-row df ac sb fw mb-4 mt-3">
```
Comment (zeripath) [inline templates/repo/commits.tmpl]: At this point this probably needs to have its own classname...

File: templates/repo/commits.tmpl [hunk 2]
```diff
@@ -3,9 +3,9 @@
...
+			<div class="df ac mt-2">
```
Comment (zeripath) [inline templates/repo/commits.tmpl]: similarly.

**Review by zeripath — COMMENTED** [review]
Looks cool but I think it would better to stick a named class on these things as the string of 2-letter codes as a classes - all of which use `!important` seems likely to cause trouble. 

I do wonder about whether dropping the `New Pull Request` label from the new pull request button makes it a bit difficult to see - previously it was pretty clear how to create a PR from a branch

File: templates/repo/commits.tmpl (hunk 1 above)
Comment (silverwind) [reply templates/repo/commits.tmpl]: It already has, but it's more bytes down the wire if I write these classes in CSS, so I prefer it this way.

**Comment by silverwind** [comment]
Did a few more fixes:

- Added margin back below subfolder path
- Made branch button render correctly without JS, eliminating the layout shift when JS loads
- Remove "Add File" button in subtree view, it seems to provide no value there.

**Comment by silverwind** [comment]
> Looks cool but I think it would better to stick a named class on these things as the string of 2-letter codes as a classes - all of which use !important seems likely to cause trouble.

I like to keep the CSS small and these helper they target a single element with a single CSS rule, that can't cause issues, the indent is clear.

> I do wonder about whether dropping the New Pull Request label from the new pull request button makes it a bit difficult to see - previously it was pretty clear how to create a PR from a branch

Guess it is fine. My initial idea was to remove this button completely. But like it is now, it takes up little space so I kept it.

**Comment by silverwind** [comment]
> So unfortunately there's still a way to get this to totally break:
> 
> Screenshot from 2022-08-01 21-39-40

Hmm, I guess long branch name can be ellipsed there to prevent it.

**Comment by silverwind** [comment]
Added truncation for long branch name:

Screen Shot 2022-08-01 at 23 36 42
Screen Shot 2022-08-01 at 23 36 15

File: templates/repo/commits.tmpl (hunk 1 above)
Comment (silverwind) [reply templates/repo/commits.tmpl]: I think this is fine, I could add another 100 bytes of CSS or keep 10 bytes in HTML, I prefer the latter.

**Comment by silverwind** [comment]
Another slightly related fix included: When clicking the clone URL input, the whole URL is selected. Sometimes one only wants to select part of the URL. I changed the handle to `focus` so manual selection after first click is still possible and this matches GH behaviour as well.

File: templates/repo/commits.tmpl (hunk 1 above)
Comment (wxiaoguang) [reply templates/repo/commits.tmpl]: I do not like these abbr names. There was a similar discussion.

https://github.com/go-gitea/gitea/pull/17898#issuecomment-986260674

ps: not blocker, just share my opinions. 😁  free free to mark this as resolved after reading.

**Review by wxiaoguang — APPROVED** [review]
LGTM (except there is a small conflict)

File: templates/repo/commits.tmpl (hunk 1 above)
Comment (silverwind) [reply templates/repo/commits.tmpl]: We could always replace with more verbose names later (maybe tailwind's, but tailwind purging can not really work with go templates I think, so I'm more leaning towards a small helper-only library if it exists).

**Comment by silverwind** [comment]
Rebased, squashed and added one more commit which brings margins in-line and added a few css rules for wiki only.

File: templates/repo/branch_dropdown.tmpl [hunk 3]
```diff
@@ -1,26 +1,27 @@
...
 			<span class="text">
 				{{if $release}}
 					{{.root.locale.Tr "repo.release.compare"}}
 				{{else}}
-					<span :class="{visible: isViewTag}" v-if="isViewTag" v-cloak>{{svg "octicon-tag"}} {{.root.locale.Tr "repo.tag"}}:</span>
-					<span :class="{visible: isViewBranch}" v-if="isViewBranch" v-cloak>{{svg "octicon-git-branch"}} {{.root.locale.Tr "repo.branch"}}:</span>
-					<span :class="{visible: isViewTree}" v-if="isViewTree" v-cloak>{{svg "octicon-git-branch"}} {{.root.locale.Tr "repo.tree"}}:</span>
+					<span :class="{visible: isViewTag}" v-if="isViewTag" {{if not (eq $type "tag")}}v-cloak{{end}}>{{svg "octicon-tag"}} {{.root.locale.Tr "repo.tag"}}:</span>
+					<span :class="{visible: isViewBranch}" v-if="isViewBranch" {{if not (eq $type "branch")}}v-cloak{{end}}>{{svg "octicon-git-branch"}} {{.root.locale.Tr "repo.branch"}}:</span>
+					<span :class="{visible: isViewTree}" v-if="isViewTree" {{if not (eq $type "tree")}}v-cloak{{end}}>{{svg "octicon-git-branch"}} {{.root.locale.Tr "repo.tree"}}:</span>
```
Comment (wxiaoguang) [inline templates/repo/branch_dropdown.tmpl:24]: A hint for reviewers, although it seems strange to use more Go template code in the Vue template, I think this is done by purpose to reduce the UI flicker 😁

**Review by wxiaoguang — APPROVED** [review]

**Review by lafriks — APPROVED** [review]

File: templates/repo/branch_dropdown.tmpl (hunk 3 above)
Comment (silverwind) [reply templates/repo/branch_dropdown.tmpl:24]: Yes, this eliminates the flash of wrong content on the branch button during initial render.

**Comment by silverwind** [comment]
Maybe should wait a bit until we have a few translations before backporting.

**Comment by silverwind** [comment]
Shoul be ready to merge, CI failure is unrelated.

**Comment by lafriks** [comment]
:rocket:

**Comment by lafriks** [comment]
🚀

**Comment by lafriks** [comment]
please backport

**Comment by silverwind** [comment]
Release branches don't receive translation updates, so I think it'd be best to wait a bit until `add_file` has some translations on `main`, then backport along with those.

**Pull request merged**